### PR TITLE
Add Golangci-lint and fix errors

### PR DESCRIPTION
**Pull request by darkowlzz**
Adds golangci-lint github action.

### CODE REVIEW

File: cmd/ignited/cmd/gitops.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ type gitOpsFlags struct {
 	username     string
 	password     string
 
-	paths []string
+	// paths []string
```
Comment (darkowlzz) [inline cmd/ignited/cmd/gitops.go:30]: This is not used anywhere. Remove it? or leave it commented?

File: pkg/filter/meta.go [hunk 2]
```diff
@@ -11,9 +11,8 @@ import (
 )
 
 const (
-	filterSeparator   = ","
-	filterApplyFailed = "failed to apply filtering"
-	regexString       = `^(?P<key>{{(?:\.|[a-zA-Z0-9]+)+}})(?P<operator>(?:=|==|!=|=~|!~))(?P<value>[a-zA-Z0-9-_:/\.\s]+)$`
+	filterSeparator = ","
+	regexString     = `^(?P<key>{{(?:\.|[a-zA-Z0-9]+)+}})(?P<operator>(?:=|==|!=|=~|!~))(?P<value>[a-zA-Z0-9-_:/\.\s]+)$`
```
Comment (darkowlzz) [inline pkg/filter/meta.go:15]: `filterApplyFailed` wasn't used anywhere.

File: pkg/source/docker.go [hunk 3]
```diff
@@ -15,7 +14,7 @@ import (
 type DockerSource struct {
 	imageRef    meta.OCIImageRef
 	cleanupFunc func() error
-	exportCmd   *exec.Cmd
+	// exportCmd   *exec.Cmd
```
Comment (darkowlzz) [inline pkg/source/docker.go:17]: This is unused. Remove it? Or keep it and add a nolint tag for it?

File: cmd/ignited/cmd/gitops.go (hunk 1 above)
Comment (twelho) [reply cmd/ignited/cmd/gitops.go:30]: I'd say remove it if it's indeed completely unused.

File: pkg/source/docker.go (hunk 3 above)
Comment (twelho) [reply pkg/source/docker.go:17]: Remove it.

File: pkg/dmlegacy/image_format.go [hunk 4]
```diff
@@ -82,7 +82,10 @@ func addFiles(img *api.Image, src source.Source) error {
 	if _, err := util.ExecuteCommand("mount", "-o", "loop", p, tempDir); err != nil {
 		return fmt.Errorf("failed to mount image %q: %v", p, err)
 	}
-	defer util.ExecuteCommand("umount", tempDir)
+	defer util.DeferErr(&err, func() error {
+		_, err := util.ExecuteCommand("umount", tempDir)
```
Comment (luxas) [inline pkg/dmlegacy/image_format.go:86]: this seems a bit dangerous, to use the same `err` variable name

Comment (darkowlzz) [reply pkg/dmlegacy/image_format.go:86]: Right, I'll change it. Thanks.

**Pull request merged**